engine/testabilities/fixture_engine.go

```diff
@@ -163,28 +174,82 @@ func (f *engineFixture) Tx() txtestability.TransactionSpec {
 
 // prepareDBConfigForTests creates a new connection that will be used as connection for engine
 func (f *engineFixture) prepareDBConfigForTests() {
-	require.Equal(f.t, datastore.SQLite, f.config.Db.Datastore.Engine, "Other datastore engines are not supported in tests (yet)")
-
-	// It is a workaround for development purpose to check the code with postgres instance.
-	if ok, dbName := testmode.CheckPostgresMode(); ok {
-		f.config.Db.Datastore.Engine = datastore.PostgreSQL
-		f.config.Db.SQL.User = "postgres"
-		f.config.Db.SQL.Password = "postgres"
-		f.config.Db.SQL.Name = dbName
-		f.config.Db.SQL.Host = "localhost"
+	// check for PostgreSQL container mode
+	if testmode.CheckPostgresContainerMode() {
```

Comment:
you're doing this here and in the line 119, it's pointless, you should remove the if in line 119